src/crimson/os/seastore/cache.h

```diff
@@ -926,20 +928,33 @@ class Cache {
 #else
     auto results = epm.alloc_new_data_extents(t, T::TYPE, length, hint, gen);
```

Comment:
> How about maintaining `max_data_allocation_size` in EPM instead, dropping `data_extents_group_t` instead of changing the return type throughout to the object-data-handler ?

I think moving `max_data_allocation_size` into EPM won't avoid "changing the return type throughout to the object-data-handler", because we have to return multiple extents instead of a single one either way, am I missing something?

Reply:
Not quite following ... `EPM::alloc_new_data_extents()` is already returning multiple extents, and so does the previous version of `Cache::alloc_new_data_extents`.

I meant the complexity of `data_extents_group_t` seems not needed if EPM is aware of `max_data_allocation_size` when preparing the buffers. Is that reasonable?